src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import './Carousel.scss';
+
+type Props = {
+  lengthMove: number;
+  setLengthMove: (lengthMove: number) => void;
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+};
+
+export const Carousel: React.FC<Props> = ({
+  lengthMove,
+  setLengthMove,
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const moveSlider = (event: React.MouseEvent<HTMLElement>) => {
+    const currentButton = event.target;
+    const buttonNext = document
+      .querySelector<HTMLButtonElement>('[data-cy=next]');
+    const buttonPrev = document
+      .querySelector<HTMLButtonElement>('[data-cy=prev]');
+    const totalWidth = images.length * itemWidth;
+    const totalFrameSize = frameSize * itemWidth;
+    const totalStep = step * itemWidth;
+
+    if (currentButton instanceof HTMLElement
+      && buttonNext instanceof HTMLElement
+      && buttonPrev instanceof HTMLElement) {
+      switch (currentButton.dataset.cy) {
+        case 'next':
+
+          if (infinite
+            && Math.abs(lengthMove) === totalWidth - totalFrameSize) {
+            setLengthMove(0);
+            break;
+          } else if (!infinite
+            && Math.abs(lengthMove) === totalWidth - totalFrameSize) {
+            buttonNext.disabled = true;
+            break;
+          }
+
+          setLengthMove(lengthMove - totalStep);
+
+          if (Math.abs(lengthMove) > totalWidth - totalFrameSize * 2) {
+            setLengthMove(-1 * (totalWidth - totalFrameSize));
+            buttonPrev.disabled = false;
+          } else {
+            buttonPrev.disabled = false;
+          }
+
+          break;
+
+        case 'prev':
+          if (infinite && lengthMove === 0) {
```

Comment:
```suggestion
          if (infinite && !lengthMove) {
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**